### DISCUSSION TITLE
Wrong decoding of negative setback values in `Climate Control Schedule CC`

### DISCUSSION THREAD

**Discussion by bueeler** [zwave-js/zwave-js#7362] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

I am using MT02650 TRV and use "Climate Control Schedule CC" command class to adjust the TRV set point based on input from room thermostat.
After sending negative setback values through function setOverride() from "Climate Control Schedule CC" command class, the value for "Override state" is not correctly shown in Z-Wave JS UI:
![grafik](https://github.com/user-attachments/assets/ad8d06b3-eab9-415c-9885-00bb3c12e187)

I would expect that negative setback values are correctly displayed as negative number in Z-Wave JS UI (as it is done for positive values). e.g. value "252" from screenshot above shall be displayed as "-0.4".

Issue seems caused by missing handling for negative numbers in function decodeSetbackState():
https://github.com/zwave-js/node-zwave-js/blob/master/packages/cc/src/lib/serializers.ts#L31
The specification defines the value as signed byte:
![grafik](https://github.com/user-attachments/assets/5d56bcb0-c358-4758-ba1a-322c373e40d7)
but the implementation treats it as "number".
All negative values where val>=128 (0x80) are handled as "special state", instead handled as negative setback state.

### Software versions

zwave-js-ui: 9.25.0.38e48a8
zwave-js: 13.10.1

### Z-Wave Controller (Stick/Dongle/...)

It should be unrelated to controller

### Device information

Manufacturer: Devolo
Model name: ...
Node ID: e.g. 008, 005 ...

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave-js-ui-store.zip](https://github.com/user-attachments/files/17629519/zwave-js-ui-store.zip)

**Comment by AlCalzone** [comment]
I was confused for a moment there - I fixed this for the `ThermostatSetbackCC` in 13.7.0, but the issue is about `Climate Control Schedule CC`.
The problem happens before the number is passed to `decodeSetbackState`, in the CC parsing routine.

**Reply by AlCalzone** [reply, marked as answer]
Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/7364
